### ISSUE TITLE
Use of TFRecord for object Detection API

### ISSUE THREAD

**Issue by hardik22317** [tensorflow/models#8271]
Here for object detection training, first we are reading the images and XML and we are encoding it and storing it in tfrecord format, then while training, we are again decoding the data and giving that data for training, then what is the use of TFRecord here? Here, actually we are making the process slower, so why can't we just give the normal data to training? Can someone please help me clear this question?

**Comment by vijaygill** [comment]
@hardik22317 I asked similar question (#8123) some time ago :).

**Comment by saeed68gm** [comment]
I agree. It's creating a lot of hard drive space issues and memory issues as @vijaygill  mentioned in #8123. Can we just use the original image blobs or create our own data reader somehow?

**Comment by GotG** [comment]
Training in Colab is very slow for SSD v2. With 5 data augmentations it takes about 6 seconds per step. I was wondering about the need to use TFRecord. Also is there any way to use caching in this case?

**Comment by muhammadsyahyas** [comment]
I am facing same issue here. I am going to train with about 600GB+ of dataset and think back about the TFRecord creation, since I have a limited disk space.

**Comment by pkulzc** [comment]
See https://www.tensorflow.org/tutorials/load_data/tfrecord

> To read data efficiently it can be helpful to serialize your data and store it in a set of files (100-200MB each) that can each be read linearly. This is especially true if the data is being streamed over a network. This can also be useful for caching any data-preprocessing.

We don't really have cycles to support custom data loader which may or may not be more efficient. But I would be happy to review and advice if anyone do make it work (aka faster).
